pydis_site/apps/api/models/bot/filters.py

```diff
@@ -0,0 +1,260 @@
+from django.contrib.postgres.fields import ArrayField
+from django.core.validators import MinValueValidator
+from django.db import models
+from django.db.models import UniqueConstraint
+
+# Must be imported that way to avoid circular imports
+from pydis_site.apps.api.models.mixins import ModelReprMixin, ModelTimestampMixin
+from .infraction import Infraction
+
+
+class FilterListType(models.IntegerChoices):
+    """Choice between allow or deny for a list type."""
+
+    ALLOW = 1
+    DENY = 0
+
+
+class FilterList(ModelTimestampMixin, ModelReprMixin, models.Model):
+    """Represent a list in its allow or deny form."""
+
+    name = models.CharField(max_length=50, help_text="The unique name of this list.")
+    list_type = models.IntegerField(
+        choices=FilterListType.choices,
+        help_text="Whether this list is an allowlist or denylist"
+    )
+    dm_content = models.CharField(
+        max_length=1000,
+        null=False,
+        blank=True,
+        help_text="The DM to send to a user triggering this filter."
+    )
+    dm_embed = models.CharField(
+        max_length=2000,
+        help_text="The content of the DM embed",
+        null=False,
+        blank=True
+    )
+    infraction_type = models.CharField(
+        choices=[
+            (choices[0].upper(), choices[1])
+            for choices in [("NONE", "None"), *Infraction.TYPE_CHOICES]
+        ],
+        max_length=10,
+        null=False,
+        help_text="The infraction to apply to this user."
+    )
+    infraction_reason = models.CharField(
+        max_length=1000,
+        help_text="The reason to give for the infraction.",
+        blank=True,
+        null=False
+    )
+    infraction_duration = models.DurationField(
+        null=False,
+        help_text="The duration of the infraction. Null if permanent."
+    )
```

Comment:
"null if permanent" is impossible given "null=False`

I see the difficulty since for `FilterBase` null means "not overwritten" so we can't really make use of it for other meanings. I guess you've already sorted this on the bot side so the help text just needs changing

Reply:
Good spot, should be 0. 4eecda9